Documentation/Proposals/NNNN-public-issues-api.md

```diff
@@ -0,0 +1,145 @@
+# Public Issues API
+
+* Proposal: [SWT-NNNN](NNNN-public-issues-api.md)
+* Authors: [Rachel Brindle](https://github.com/younata)
+* Status: **Awaiting implementation**
+* Bug: [apple/swift-testing#474](https://github.com/apple/swift-testing/issues/474)
+* Implementation: n/a
+* Review: n/a
+
+## Introduction
+
+Swift Testing should provide a more robust Issue-reporting API, to facilitate
+better integration with third-party assertion tools.
+
+## Motivation
+
+There are several third-party assertion libraries that developers are used to
+using, and Swift Testing should make it as easy as possible for the developers
+and maintainers of these third-party assertion libraries to integrate with
+Swift Testing.
+
+Swift Testing currently offers the `Issue.record` family of static methods,
+which provides an acceptable interface for one-off/custom assertions without
+using the `#expect` or `#require` macros. Unfortunately, `Issue.record` is not
+robust enough for dedicated assertion libraries like Nimble. For example,
+there is no current way to provide an `Issue.Kind` to `Issue.record`.
+
+
+## Proposed solution
+
+I propose making the initializer for `Issue` public, and provide a public
+`Issue.record()` instance method to report that Issue. This provides a clean,
+easy to maintain way to create and record custom Issues that assertion tools
+are already used to because it is analogous to the XCTIssue API in XCTest.
+
+## Detailed design
+
+Create a public initializer for `Issue`, which is not gated by
+`@_spi(ForToolsIntegrationOnly)`:
+
+```swift
+public struct Issue: Sendable {
+  // ...
+
+  /// Initialize an issue instance with the specified details.
+  ///
+  /// - Parameters:
+  ///   - kind: The kind of issue this value represents.
+  ///   - comments: An array of comments describing the issue. This array may be
+  ///     empty.
+  ///   - sourceContext: A ``SourceContext`` indicating where and how this issue
+  ///     occurred. This defaults to a default source context returned by
+  ///     calling ``SourceContext/init(backtrace:sourceLocation:)`` with zero
+  ///     arguments.
+  public init(
+    kind: Kind,
+    comments: [Comment],
+    sourceContext: SourceContext = .init()
+  ) {
+    self.kind = kind
+    self.comments = comments
+    self.sourceContext = sourceContext
+  }
```

Comment:
Here too, you can omit the body of the initializer — only the signature is needed for review purposes